### PR TITLE
SAA-1256: Refactor create appointment set

### PR DESCRIPTION
**Pull request by iterative-it**
**MEDIUM RISK**

Tested locally and will test on dev once there. This PR also improves test coverage of the appointment set creation process, something that was light before.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentCreateDomainService.kt [hunk 1]
```diff
@@ -95,3 +77,28 @@ class AppointmentCreateDomainService(
...
+    onWing = this.onWing,
+    offWing = this.offWing,
+    startDate = startDate,
+    startTime = this.startTime,
+    endTime = this.endTime,
+    unlockNotes = this.unlockNotes,
+    extraInformation = this.extraInformation,
+    createdTime = this.createdTime,
+    createdBy = this.createdBy,
+    updatedTime = this.updatedTime,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentCreateDomainService.kt:102]: Going by the name of the function here 'create'.  Could update time and updated by be populated here?  This is a create.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentSetService.kt [hunk 2]
```diff
@@ -40,4 +65,132 @@ class AppointmentSetService(
...
+    request.failIfMissingPrisoners(prisonNumberBookingIdMap)
+
+    val appointmentTier = appointmentTierRepository.findOrThrowNotFound(NOT_SPECIFIED_APPOINTMENT_TIER_ID)
+
+    return transactionHandler.newSpringTransaction {
+      appointmentSetRepository.saveAndFlush(
+        request.toAppointmentSet(prisonNumberBookingIdMap, appointmentTier, principal.name),
+      )
+    }.also {
+      // TODO: publish appointment instance created messages post transaction
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentSetService.kt:86]: Is there a ticket to finish this TODO off?

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentCreateDomainService.kt (hunk 1 above)
Comment (iterative-it) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/AppointmentCreateDomainService.kt:102]: Good point. This is to support creating migrated appointments that could have been updated in NOMIS/DPS. It's to ensure that their state is transferred over. It's confusing though so I will add a code comment

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentSetService.kt (hunk 2 above)
Comment (iterative-it) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/AppointmentSetService.kt:86]: There is, SAA-1258

**Pull request merged**